creating_citation_counts_tsv/scripts/summarize_gene_citations_all_species.py

```diff
@@ -0,0 +1,335 @@
+""" 
+Creating TSVs containing the 10 most-cited genes per species and their gene information.
+"""
+import os
+import sys
+import json
+import pyspark
+import requests
+from operator import add
+from datetime import date, timedelta, datetime
+import csv
+from collections import Counter
+
+
+def getting_recent_gene2pubmed(species):
+    """ 
+    Getting a list of the gene ID, gene publication citation ID, and count of how frequently the gene is cited over the past 5 months for each gene.
+    """
+
+    # Getting a list of the gene ID and gene publication citation ID pairs and saving it to the gene2pubmed variable 
+    gene2pubmed = (sc.textFile(f"creating_citation_counts_tsv/data/{species}/gene2pubmed")
+                    .filter(lambda x: x[0] != '#')
+                    .map(lambda x: x.split('\t'))
+                    .map(lambda x: ( str(x[2]), int(x[1]))))
+    
+    # Outputing the first element in the gene2pubmed variable
+    print(gene2pubmed.take(1))
+    
+    # Getting the list of unique gene publication citation IDs
+    pubmeds_set = set([x[0] for x in gene2pubmed.collect()])
+
+    # Getting the list of over gene publication citation IDs the past five months and saving it to the year_pmid variable 
+    year_pmid = (sc.textFile("creating_citation_counts_tsv/data/recent_pmid_year.ssv")
+                    .map(lambda x: x.split())
+                    .map(lambda x: (str(x[1]), int(x[0])))
+                    .filter(lambda x: x[0] in pubmeds_set))
+    
+    # Outputing the first element in the year_pmid variable
+    print(year_pmid.take(1))
+    
+    # Getting a list of the gene ID and gene publication citation ID pairs over the past 5 months and saving it to the gene2pubmed_to_year variable
+    gene2pubmed_to_year = gene2pubmed.join(year_pmid).map(lambda x: (x[1][0], x[0]))
+    
+    # Outputing the first element in the gene2pubmed_to_year variable
+    print(gene2pubmed_to_year.take(1))
+    
+    # Getting a list of the gene ID, gene publication citation ID, count of how frequently the gene is cited over the past 5 months and saving it to the gene_counts variable
+    gene_counts = gene2pubmed.join(year_pmid).map(lambda x: (x[1][0], 1)).reduceByKey(add).join(gene2pubmed_to_year).map(lambda x: ( str(x[0]), (x[1][1] ,x[1][0])))
```

Comment:
For readability, perhaps split this one complex statement into a few simpler ones, or at least one complex statement on multiple lines (like your assignment of `year_pmid` above).

Reply:
I scared it might be a lot of memory in RAM, if make separate variables